classes/db/Db.php

```diff
@@ -567,10 +567,10 @@ public function executeS($sql, $array = true, $use_cache = true)
 		if ($use_cache && $this->is_cache_enabled && $array)
 		{
 			$this->last_query_hash = Tools::encryptIV($sql);
-			if (($result = Cache::getInstance()->get($this->last_query_hash)) !== false)
+			if (Cache::getInstance()->exists($this->last_query_hash) !== false)
```

Comment:
Well actually I would prefer to just add the isset($this->keys[$key]) is the _get() function of CacheFS. It would cost almost nothing, and actually it needs be done in both _exists() & _get() before the 
if ($this->keys[$key] > 0 && $this->keys[$key] < time())
test to avoid a lot of php notice.